brave/brave-core
# Refactors database

Resolves https://github.com/brave/brave-browser/issues/2163
Resolves https://github.com/brave/brave-browser/issues/2551

Native ledger: https://github.com/brave-intl/bat-native-ledger/pull/183

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
- general regression of rewards
- focused on adding things to table and contribution
- make sure that upgrade path (form previous version) is working correctly, that no data is lost
- delete some sites and then restore them, make sure that excluded num is tracked correctly
- try doing ac and make sure that all items from the table are send out

## Reviewer Checklist:
